include/ginkgo/core/matrix/bccoo.hpp

```diff
@@ -0,0 +1,620 @@
+
+    /**
+     * @copydoc Bccoo::get_chunk()
+     *
+     * @note This is the constant version of the function, which can be
+     *       significantly more memory efficient than the non-constant version,
+     *       so always prefer this version.
+     */
+    const uint8* get_const_chunk() const noexcept
+    {
+        return chunk_.get_const_data();
+    }
+
+    /**
+     * Returns the number of elements explicitly stored in the matrix.
+     *
+     * @return the number of elements explicitly stored in the matrix
+     */
+    index_type get_num_stored_elements() const noexcept
+    {
+        return num_nonzeros_;
+    }
+
+    /**
+     * Returns the block size used in the definition of the matrix.
+     *
+     * @return the block size used in the definition of the matrix.
+     */
+    index_type get_block_size() const noexcept { return block_size_; }
+
+    /**
+     * Returns the number of blocks used in the definition of the matrix.
+     *
+     * @return the number of blocks used in the definition of the matrix.
+     */
+    index_type get_num_blocks() const noexcept { return rows_.get_num_elems(); }
+
+    /**
+     * Returns the number of bytes of chunk vector used in the definition of the
+     * matrix.
+     *
+     * @return the number of bytes of chunk vector used in the definition of the
+     * matrix.
+     */
+    size_type get_num_bytes() const noexcept { return chunk_.get_num_elems(); }
+
+    /**
+     * Returns the compression used in the definition of the matrix.
+     *
+     * @return the compression used in the definition of the matrix.
+     */
+    bccoo::compression get_compression() const noexcept { return compression_; }
+
+    /**
+     * Returns if the default compression is used
+     *
+     * @returns if the default compression is used
+     */
+    bool use_default_compression() const noexcept
+    {
+        return compression_ == bccoo::compression::def_value;
+    }
+
+    /**
+     * Returns if the element compression is used
+     *
+     * @returns if the element compression is used
+     */
+    bool use_element_compression() const noexcept
+    {
+        return compression_ == bccoo::compression::element;
+    }
+
+    /**
+     * Returns if the block compression is used
+     *
+     * @returns if the block compression is used
+     */
+    bool use_block_compression() const noexcept
+    {
+        return compression_ == bccoo::compression::block;
+    }
+
+    /**
+     * Returns if the object is initialized
+     *
+     * @returns if the object is initialized
+     */
+    bool is_initialized() const
+    {
+        //        return (this->use_default_compression() != true &&
+        //        this->get_block_size() > 0;
+        return (compression_ != bccoo::compression::def_value &&
+                block_size_ > 0);
+    }
+
+    // TODO: It could be better to use
+    // TODO:        b = Bccoo * x + b
+    // TODO: It would be easier to understand
```

Comment:
I agree, but if we use the same variable names in other classes as well, sticking to is is more important.

Reply:
Done!!